mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js

```diff
@@ -70,7 +149,27 @@ AUI.add(
 							todoList.append(taskHtml);
 
 							taskInput.val('');
+
+							instance._countTasks();
 						}
+					},
+
+					_countTasks: function() {
+						var instance = this;
+
+						var todoList = instance._todoList;
+
+						var taskListLi = todoList.all('li');
+
+						var remainingCount = taskListLi.size();
+
+						var remainingDiv = instance.byId('remaining-tasks-count');
```

Comment:
You are making unnecessary DOM queries within a click event handler. As I mentioned before, if the queried elements never change you should query then only once in ```initializer()``` and store it as ```instance``` property. See ```instance._appendTodoList()```. The only case when DOM queries from event handler make sense is if you are querying element relative to click location, like on line 111.